File: apps/ios/Inkuna/Reader/ReaderPagerSurface.swift
import UIKit

/// The engine-shaped strip contract shared by the native reader pagers.
@MainActor
protocol ReaderPagerSurface: AnyObject {
    var isEngageable: Bool { get }
    var isBusy: Bool { get }
    var hasActiveSelection: Bool { get }
    var isRightToLeft: Bool { get }
    func innerMetrics() -> ReaderPagerStrip?
    func setInnerOffset(_ x: CGFloat)
    func outerMetrics() -> ReaderPagerStrip?
    func setOuterOffset(_ x: CGFloat)
    func neighborIsReady(toRight: Bool) -> Bool
    func commitBoundaryCrossing(toRight: Bool) -> Bool
}

struct ReaderPagerStrip {
    var offset: CGFloat
    var range: ClosedRange<CGFloat>
    var pageWidth: CGFloat
}

File: apps/android/app/src/main/java/app/inkuna/android/ui/reader/ReaderPagerSurface.kt
package app.inkuna.android.ui.reader

/**
 * The renderer-shaped strip contract shared by the Android reader pagers.
 *
 * Offsets are view pixels: the pager consumes touch coordinates, while page
 * display-list geometry remains in layout points inside [engine.PageView].
 */
interface ReaderPagerSurface {
    val isEngageable: Boolean
    val isBusy: Boolean
    val hasActiveSelection: Boolean
    val isRightToLeft: Boolean

    fun innerMetrics(): ReaderPagerStrip?
    fun setInnerOffset(x: Float)
    fun outerMetrics(): ReaderPagerStrip?
    fun setOuterOffset(x: Float)
    fun neighborIsReady(toRight: Boolean): Boolean
    fun commitBoundaryCrossing(toRight: Boolean): Boolean
}

data class ReaderPagerStrip(
    val offset: Float,
    val range: ClosedFloatingPointRange<Float>,
    val pageWidth: Float,
)
